app/flatpak-builtins-info.c

```diff
@@ -180,6 +180,26 @@ flatpak_builtin_info (int argc, char **argv, GCancellable *cancellable, GError *
       g_autofree char *latest = NULL;
       const char *xa_metadata = NULL;
       const char *collection_id = NULL;
+      int len;
+      int rows, cols;
+      int width;
+      g_autoptr(AsStore) store = as_store_new ();
+      g_autoptr(GError) local_error = NULL;
+      AsApp *app = NULL;
+
+      if (!flatpak_dir_load_appstream_store (dir, origin, parts[2], store, NULL, &local_error))
+        {
+          g_warning ("%s", local_error->message);
+          g_clear_error (&local_error);
+        }
+
+      app = as_store_get_app_by_id (store, parts[1]);
+      if (app)
+        {
+          const char *name = as_app_get_localized_name (app);
+          const char *comment = as_app_get_localized_comment (app);
+          g_print ("\n%s - %s\n\n", name, comment);
+        }
```

Comment:
I think we should get the appdata from the deployed appdata instead of the remote appstream (i.e. ~/.local/share/flatpak/app/org.gnome.gedit/x86_64/stable/active/files/share/appdata/org.gnome.gedit.appdata.xml or equivalent), because the appstream might be different than the version that is deployed. I mean, its definately true if we want to show the version string from the appdata.

Reply:
this is quite the disaster area

there's appstream/, app-info/, appdata/, metainfo/ all with various xml-ish or gzipped files and whatnot

different apps seem to have different things

and the libappstream api is supremely unhelpful in figuring out what to load how

I need to consult richard to know what can be relied on